src/provider/provider_tracking.c

```diff
@@ -655,7 +767,8 @@ static umf_result_t upstreamOpenIPCHandle(umf_tracking_memory_provider_t *p,
     }
     assert(mapped_ptr != NULL);
 
-    ret = umfMemoryTrackerAdd(p->hTracker, p->pool, mapped_ptr, bufferSize);
+    ret = umfMemoryTrackerAddIpcSegment(p->hTracker, mapped_ptr, bufferSize,
+                                        p->pool->provider, cache_entry);
```

Comment:
Does `p->pool->provider` break encapsulation?
Should we expose UMF internals to the provider's implementation?

Reply:
I consider the tracking provider as an internal part because it is not a public provider it is just a memory-tracking utility implemented as a provider for our convenience.

But it is a good catch. Today we always create a `pool` and get an IPC handler from the pool. And tracking provider is always created together with a pool.  And the IPC handler is a tracking memory provider in fact.
However, we are considering adding an API that allows IPC handler creation from scratch (without creating a memory pool). In that case, we will create a tracker provider without a pool. So `p->pool` might be `NULL` in the future...

Let me think more about that.

Reply:
I added a TODO comment. it is just an implementation detail that can be changed in the future.